projects/lights_control/src/lights_can.c

```diff
@@ -0,0 +1,109 @@
+#include "can.h"
+#include "can_hw.h"
+#include "can_msg_defs.h"
+
+#include "can_unpack_impl.h"
+#include "event_queue.h"
+#include "status.h"
+
+#include "lights_can.h"
+#include "lights_events.h"
+#include "lights_gpio.h"
+
+#define SOMEPORT 0
+#define SOMEPIN 0
+
+#define CASE_ACTION(action)            \
+  case ACTION_##action:                \
+    event_raise(EVENT_##action, data); \
+    break;
+
+#define CAN_RX_ADDR \
+  { 0, 11 }
+
+#define CAN_TX_ADDR \
+  { 0, 12 }
+
+// TODO(ELEC-372): remove these and get the device ID from elsewhere
+#define CAN_DEVICE_ID_FRONT 0
+#define CAN_DEVICE_ID_REAR 0
+
+static BoardType s_boardtype;
+
+static CANStorage s_can_storage;
+static CANRxHandler s_rx_handlers[CAN_NUM_RX_HANDLERS];
+
+static StatusCode prv_rx_handler(const CANMessage *msg, void *context, CANAckStatus *ack_reply);
+
+// TODO(ELEC-372):
+// need to figure out:
+//      1.bitrate for can_settings
+//      2.CAN_TX_ADDR
+//      3.CAN_RX_ADDR
+//      4.msg_id
+void lights_can_init(BoardType boardtype, bool loopback) {
+  CANSettings can_settings = { .bitrate = CAN_HW_BITRATE_125KBPS,
```

Comment:
Maybe move this struct to a config file and pass it to init as a const pointer? It is best to assign all the ports and pins in one location so it is easy to change the if we ever update the hardware. Also it avoids leaking the loopback boolean everywhere.

Reply:
yeah good point, I created a `lights_config.h` header file which contains this enum, thanks.